cmake/mcuboot.cmake

```diff
@@ -0,0 +1,107 @@
+# CONFIG_MCUBOOT_SIGNATURE_KEY_FILE is set to a nonempty value.
+#
+# It builds signed binaries using imgtool as a post-processing step
+# after zephyr/zephyr.elf is created in the build directory.
+#
+# Since this file is brought in via include(), we do the work in a
+# function to avoid polluting the top-level scope.
+
+function(zephyr_mcuboot_tasks)
+  if(CONFIG_MCUBOOT_SIGNATURE_KEY_FILE STREQUAL "")
+    # No signature key file, no signed binaries. No warning is needed;
+    # this is the documented behavior.
+    return()
+  elseif(NOT EXISTS "${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}")
+    message(WARNING "Not signing images for MCUboot: CONFIG_MCUBOOT_SIGNATURE_KEY_FILE=\"${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}\" does not exist.")
+    return()
+  elseif(NOT WEST)
+    message(WARNING "Not signing images for MCUboot: west not found.")
+    return()
+  endif()
+
+  # Even though west is installed, imgtool might not be if the user has
+  # a custom manifest which doesn't include MCUboot.
+  #
+  # Therefore, go with an explicitly installed imgtool first, falling
+  # back on mcuboot/scripts/imgtool.py.
+  if(IMGTOOL)
+    set(imgtool_path "${IMGTOOL}")
+  else()
+    # We can't use ZEPHYR_MCUBOOT_MODULE_DIR because mcuboot is not a
+    # "real" zephyr module. It contains no CMakeLists.txt or Kconfig
+    # files. At time of writing, its zephyr/module.yml file was just for
+    # sanitycheck.
+    #
+    # Therefore, the build system doesn't regard it as a module, and
+    # doesn't define ZEPHYR_MCUBOOT_MODULE_DIR.
+    execute_process(COMMAND ${WEST} list -f "{posixpath}" mcuboot
+      WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}"
+      RESULT_VARIABLE out
+      OUTPUT_VARIABLE mcuboot_location
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(out EQUAL 0)
+      set(IMGTOOL_PY "${mcuboot_location}/scripts/imgtool.py")
+      if(EXISTS "${IMGTOOL_PY}")
+        set(imgtool_path "${IMGTOOL_PY}")
+      endif()
+    endif()
+  endif()
+
+  # No imgtool, no signed binaries.
+  if(NOT DEFINED imgtool_path)
+    message(WARNING "Not signing images with imgtool: can't find imgtool. To fix, install imgtool with pip3 and make sure it's in PATH, or add the mcuboot repository to the west manifest")
+    return()
+  endif()
+
+  if(NOT CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS STREQUAL "")
+    # We need to separate any extra arguments into a CMake list so they
+    # can be added to POST_BUILD_COMMANDS as a properly formatted ;-list.
+    #
+    # Do so using UNIX_COMMAND syntax for uniformity, allowing for
+    # consistent results regardless of the host platform.
+    separate_arguments(imgtool_extra UNIX_COMMAND ${CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS})
+  else()
+    set(imgtool_extra)
+  endif()
+
+  # Put the west sign outputs in their current default locations
+  # explicitly, in case those defaults change.
+
+  if(CONFIG_BUILD_OUTPUT_BIN)
+    set(bin_args --bin --sbin ${ZEPHYR_BINARY_DIR}/zephyr.signed.bin)
```

Comment:
Does it makes sense for the output binary to be ` ${ZEPHYR_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.signed.bin`?
Default behaviour would still be `zephyr.bin` and `zephyr.signed.bin`, but `custom_app_name.bin` would generate `custom_app_name.signed.bin`.
Same reasoning would apply to the hex output.

Reply:
I agree.

Reply:
> Does it makes sense for the output binary to be ${ZEPHYR_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.signed.bin?
> Same reasoning would apply to the hex output.

Agreed.